### PR TITLE
Gui backend integration PR

### PR DESCRIPTION
**Pull request by jbcohen4**
Pull request for Gui_Backend_Integration branch. Some changes are needed before we can merge with main.

### CODE REVIEW

File: backend.py [hunk 1]
```diff
@@ -14,6 +13,21 @@ def read_file_as_string(file_path):
     except FileNotFoundError:
         return Err("File not found.")
 
+def read_values_from_file(file_path):
```
Comment (jbcohen4) [inline backend.py]: A docstring describing what the function does would be helpful. Please say the format of data you're expecting in the file. Please also describe what the function returns (if anything). Here, it looks like you're expecting the file to be in the format:
"
key : value
key : value
"

and you're returning the dictionary that file represents. It looks like all keys and all values are kept as strings.

File: backend.py [hunk 2]
```diff
@@ -14,6 +13,21 @@ def read_file_as_string(file_path):
...
+            
+            # Assuming each line contains a key-value pair separated by ':'
+            values = {}
+            for line in lines:
+                key, value = map(str.strip, line.split(':'))
+                values[key] = value
+                
+            return values
+    except FileNotFoundError:
+        return Err("File not found.")
```
Comment (jbcohen4) [inline backend.py]: Perfect! Thank you for returning an error like this.

File: backend.py [hunk 3]
```diff
@@ -148,20 +163,47 @@ def current_to_temp_for_AD590(current: float) -> float:
     Temp = Current * 10 ^ 6 - 273, where Temp is in °C, and Current is in Amps."""
     return current * 10**6 - 273
 
+# added by Rahul - as current output is required in microamps for y-axis
```
Comment (jbcohen4) [inline backend.py]: Thanks for the comment. Super helpful

File: backend.py [hunk 4]
```diff
@@ -148,20 +163,47 @@ def current_to_temp_for_AD590(current: float) -> float:
...
-            temp = current_to_temp_for_AD590(current)
-            writer.writerow([avg_fluences, temp])
+            Iout = current_to_Iout_for_AD590(current)
+            writer.writerow([avg_fluences, Iout])
 
 
 if __name__ == "__main__": # python best practice. Ask google or ChatGPT if confused.
-    write_fluences_vs_temp_to_csv('output/fluences-vs-temp.csv')
+    
+    # shared variables
```
Comment (jbcohen4) [inline backend.py]: Don't use shared variables. They should be passed as inputs to the functions that need them.

File: backend.py [hunk 5]
```diff
@@ -122,7 +136,7 @@ def generate_single_current_value(pnp_is: float, pnp_n: float, npn_is: float, np
     outtext = read_file_as_string(outfile)
     out_data = parse_output_data(outtext)
     for voltage, current in out_data:
-        if voltage == 5.0:
+        if voltage == float(input_voltage):
```
Comment (jbcohen4) [inline backend.py]: Using shared variable "input_voltage". Please don't do that. Instead, make the function take "input_voltage" as an input.

File: gui.py [hunk 6]
```diff
@@ -1,157 +1,219 @@
...
+
+    # Write values to a file
+    with open('shared-config/shared_values.txt', 'w') as file:
+        file.write(f"Voltage: {Voltage}\n")
+        file.write(f"Temperature: {Temperature}\n")
+        file.write(f"Fluence Min: {Fluence_Min}\n")
+        file.write(f"Fluence Max: {Fluence_Max}\n")
+
+    # Build the command to run backend.py
+    command = ["python", "backend.py"]
```
Comment (jbcohen4) [inline gui.py]: We should not need to execute backend.py in a subprocess. We should instead have a function in backend.py that takes the variables it needs as inputs and returns the data the gui needs.

File: gui.py [hunk 7]
```diff
@@ -1,157 +1,219 @@
...
+        range_min = float(Fluence_Min) * 10**11
+        range_max = float(Fluence_Max)  * 10**13
+        print(range_min, range_max)
+        x_range_to_include = (range_min , range_max)
+        execute_function(sample_data, x_range_to_include)
+    else:
+        execute_function(sample_data)
+
+# Execute function
+def execute_function(sample_data, x_range=None):
```
Comment (jbcohen4) [inline gui.py]: Can you leave a docstring or comment saying what this function does? The name does not tell me much.

**Review by jbcohen4 — COMMENTED** [review]
Thanks for integrating the backend with the gui. I gave some comments on code style

File: backend.py (hunk 1 above)
Comment (RahulSihara) [reply backend.py]: my bad, i did not read earlier comments as i was not aware you posted some comments. 
Thanks for the feedback.

**Pull request merged**